components/script/dom/document.rs

```diff
@@ -765,7 +786,56 @@ impl Document {
 
         if let MouseEventType::Click = mouse_event_type {
             self.commit_focus_transaction(FocusType::Element);
+
+            // https://w3c.github.io/uievents/#event-type-dblclick
+            let DBL_CLICK_TIMEOUT = Duration::from_millis(PREFS.get("dom.document.dblclick_timeout").as_u64().unwrap());
+            let DBL_CLICK_DIST_THRESHOLD = PREFS.get("dom.document.dblclick_dist").as_u64().unwrap();
+
+            let new_pos = Point2D::new(client_x, client_y);
+            let now = Instant::now();
+            let opt = self.last_click_info.borrow_mut().take();
+
+            *self.last_click_info.borrow_mut() = match opt {
+                Some((last_time, last_pos)) => {
+                    let line = new_pos - last_pos;
+                    let dist = (line.dot(line) as f64).sqrt();
+
+                    if now.duration_since(last_time) < DBL_CLICK_TIMEOUT &&
+                        dist < DBL_CLICK_DIST_THRESHOLD as f64 {
```

Comment:
I originally included this check but then realised that FF/Chrome do not care for the case when the first click is over one element and the second click (which forms the double click) is over a different element. When this happens, the dblclick event is fired over the second element regardless of the fact that the second element clicked was different to the first element. 

I am using [this](https://samuknet.github.io/test_cases/dblclick/) to compare dblclick behaviour - a pink border is toggled on any DIV which receives a `dblclick` event.

This is an example of FF behavior for the above case.
![different_element_click](https://cloud.githubusercontent.com/assets/9294710/18122142/e9e6f384-6f5f-11e6-83fe-52513df892f8.gif)

My thinking was just to mirror this behavior with the Servo code, although this may not be the officially correct thing - what do you think?